homework/index.js

```diff
@@ -33,10 +33,81 @@
+
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const header = createAndAppend('header', root, { id: 'header' });
+        createAndAppend('p', header, { text: 'HYF Repositories' });
+        const select = createAndAppend('select', header, { id: 'select' });
+
+        // CREATE EVENT LISTENER
+        select.addEventListener(
+          'change',
+          () => {
+            const selected = select.selectedIndex;
+            const repositoryValue = data[selected].name;
+            const repoLink = data[selected].html_url;
+            if (repositoryValue === null) {
+              document.getElementById('property_0').innerText = '-';
+            } else {
+              document.getElementById('property_0').innerText = repositoryValue;
+              document.getElementById('property_0').setAttribute('href', repoLink);
+              document.getElementById('property_0').setAttribute('target', '_blank');
+            }
+            const descriptionValue = data[selected].description;
+            if (descriptionValue === null) {
+              document.getElementById('property_1').innerText = '-';
+            } else {
+              document.getElementById('property_1').innerText = descriptionValue;
+            }
+            const forksValue = data[selected].forks;
+            if (forksValue === null) {
+              document.getElementById('property_2').innerText = '-';
+            } else {
+              document.getElementById('property_2').innerText = forksValue;
+            }
+            const updatedValue = new Date(data[selected].updated_at);
+            if (updatedValue === null) {
+              document.getElementById('property_3').innerText = '-';
+            } else {
+              document.getElementById('property_3').innerText = updatedValue;
+            }
+          },
+          false,
+        );
+
+        // SORT SELECTION MENU ALPHABETICALLY
+        const dataSorted = data.sort((a, b) => a.name.localeCompare(b.name));
+
+        // CREATE OPTIONS OF SELECTION MENU
+        dataSorted.forEach((repo, index) => {
+          createAndAppend('option', select, { text: repo.name, value: index });
+        });
+
+        const leftColumn = createAndAppend('div', root, { id: 'info_panel' });
+        const leftColumnTable = createAndAppend('table', leftColumn);
+        const leftColumnTableProperties = ['Repository :', 'Description :', 'Forks :', 'Updated :'];
+        const leftColumnTableFirstTr = createAndAppend('tr', leftColumnTable);
+        createAndAppend('td', leftColumnTableFirstTr, { text: leftColumnTableProperties[0] });
+        const linkRepo = createAndAppend('td', leftColumnTableFirstTr);
+        createAndAppend('a', linkRepo, { id: `property_${0}` });
+
+        // CREATE TABLE ELEMENTS OF LEFT COLUMN
+        for (let j = 1; j < 4; j++) {
```

Comment:
What if I want to add a property? Let's say, I modify the array `leftColumnTableProperties`, and I add the property `"Maintainer: "`. Will it appear in the table?

Reply:
I understand. Instead of using specific values, I could use dynamic values like array.length in for loop's definition. I changed it.